crates/devolutions-pedm/src/db/mod.rs

```diff
@@ -197,4 +199,77 @@ pub(crate) trait Database: Send + Sync {
     async fn log_http_request(&self, req_id: i32, method: &str, path: &str, status_code: i16) -> Result<(), DbError>;
 
     async fn insert_elevate_tmp_request(&self, req_id: i32, seconds: i32) -> Result<(), DbError>;
+
+    async fn insert_jit_elevation_result(&self, result: &ElevationRequest) -> Result<(), DbError>;
+}
+
+// Bridge for DB operations from synchronous functions.
+// This may or may not be a temporary workaround.
+
+#[derive(Clone)]
+pub(crate) struct DbHandle {
+    tx: tokio::sync::mpsc::Sender<DbRequest>,
+}
+
+impl DbHandle {
+    pub(crate) fn insert_jit_elevation_result(&self, result: ElevationRequest) {
+        if let Err(error) = self.tx.blocking_send(DbRequest::InsertJitElevationResult(result)) {
+            if let DbRequest::InsertJitElevationResult(result) = error.0 {
+                // We also log the elevation result here, so it’s not completely lost.
+                error!(
+                    ?result,
+                    "DB sync bridge is down, failed to insert the JIT elevation result in the database"
+                );
+            }
+        }
+    }
+}
+
+enum DbRequest {
+    InsertJitElevationResult(ElevationRequest),
+}
```

Comment:
@thenextman Add a new variant here if you need the bridge to support more operations (config, etc).